xonsh/dirstack.py

```diff
@@ -88,6 +88,8 @@ def cd(args, stdin=None):
         return '', 'cd: no such file or directory: {0}\n'.format(d)
     if not os.path.isdir(d):
         return '', 'cd: {0} is not a directory\n'.format(d)
+    if not os.access(d, os.X_OK):
+        return '', 'cd: permission denied: {0}\n'.format(d)
```

Comment:
Just an idea, but now we have the option of returning a returncode as the third element of the tuple. Maybe you could add a `, 1` to the end of this line?